android/src/main/java/com/qubit/reactnative/sdk/QubitSDKModule.java

```diff
@@ -139,18 +147,85 @@ public void getExperiences(ReadableArray experienceIds,
 
   @ReactMethod
   public void experienceShown(String callback) {
-    CallbackConnector callbackConnector = new CallbackConnectorImpl(callback, QubitSDK.getDeviceId());
+    ExperienceCallbackConnector callbackConnector = new ExperienceCallbackConnectorImpl(callback, QubitSDK.getDeviceId());
     callbackConnector.shown();
   }
 
+  @ReactMethod
+  public void getPlacement(
+      String placementId,
+      String mode,
+      String attributes,
+      String campaignId,
+      String experienceId,
+      Promise placementPromise
+  ) {
+    QubitSDK.getPlacement(
+        placementId,
+        matchMode(mode),
+        getAttributesJson(attributes),
+        new PlacementPreviewOptions(campaignId, experienceId),
+        placement -> {
+          JsonObject placementJson = placement.getContent();
+          placementJson.addProperty("impressionUrl", placement.getImpressionUrl());
+          placementJson.addProperty("clickthroughUrl", placement.getClickthroughUrl());
+          placementPromise.resolve(WritableMapConverter.convertJsonToMap(placementJson));
+          return null;
+        },
+        throwable -> {
+          placementPromise.reject(throwable);
+          return null;
+        }
+    );
+  }
+
+  @ReactMethod
+  public void placementImpression(String callbackUrl) {
+    PlacementCallbackConnector callbackConnector = new PlacementCallbackConnectorImpl(
```

Comment:
Wrapping it into PlacementCallbackConnector seems to be redundant. We don't need implementation of PlacementCallbackConnector here.

It can be replaced with simple:
```
QubitSDK.getCallbackRequestTracker().scheduleRequest(callbackUrl)
```

Reply:
Actually I followed your other suggestion - added `sendCallbackRequest(url)` method to `QubitSDK` and used it here.